src/main.rs

```diff
@@ -117,9 +125,20 @@ fn main() -> Result<()> {
     // Read configuration from disk or generate a default one.
     let config = Config::get_or_create()?;
     let _ = fs::create_dir(&config.temporary_project_dir);
-    let tmp_dir = Builder::new()
-        .prefix("tmp-")
-        .tempdir_in(&config.temporary_project_dir)?;
+
+    // Create the temporary directory
+    let tmp_dir = {
+        if cli.worktree_branch.is_some() {
+            Builder::new()
+                .prefix("tmp-")
+                .tempdir_in(&config.git_worktree_dir)?
+        } else {
+            Builder::new()
+                .prefix("tmp-")
+                .tempdir_in(&config.temporary_project_dir)?
+        }
```

Comment:
I don't see why you would want to use a different par for worktree checkouts :thinking: Can you explain the reasoning behind this?

Reply:
If you want to have your working trees in another directory instead of the one you use for your temporary project

Reply:
Why would I call "cargo temp" if I don't want the stuff to be temporary XD

Reply:
Because we have the possibility to "save" the temporary stuff. For me working trees and temporary projects are different stuffs.